src/data/markdown/docs/03 cloud/03 Manage/04 Environment variables.md

```diff
@@ -0,0 +1,95 @@
+
+> Changes to an Environment Variable won't apply to already started Test-Runs, they only apply to new upcoming Test-Runs.
+
+## Managing environment variables
+
+Environment Variables can be managed in Organization Settings: Select your profile icon, then **Manage > Environment variables**. Permission to create, reveal, modify and delete environment variables is restricted to Owners and Admins of the organization. 
+
+![k6 Environment Variable](./images/envvars/environment-variables.png)
+
+#### Naming environment variables
+
+The following rules must apply for naming an environment variable:
+- Names can only contain alphanumeric characters (`[a-z]`, `[A-Z]`, `[0-9]`) or underscores (`_`). Spaces are not allowed.
+- Names must not start with the `K6_CLOUD` prefix.
+- Names must not start with a number.
+- Names are not case-sensitive.
+- Names must be unique.
+
+#### Declaring a new environment variable
+
+Enter the Name for your Environment Variable. Then, enter the respective Value. Optionally, you can also add a brief description then click **submit**. 
+
+![k6 Environment Variable](./images/envvars/create-environment-variable.png)
+
+### Accessing environment variables
+
+The cloud Environment Variable can be referenced in the [script-editor](https://k6.io/docs/cloud/creating-and-running-a-test/script-editor/) through the global `__ENV` variable, the same way as in [k6 CLI Environment Variable](https://k6.io/docs/using-k6/environment-variables/#passing-environment-variables-to-the-k6-script)
+
+```javascript
+import http from 'k6/http';
+import { sleep } from 'k6';
+
+export default function () {
+  const params = {
+    headers: { Authorization: __ENV.AUTH_TOKEN },
+  };
+  const res = http.get(__ENV.SERVICE_URL, params);
+  sleep(1);
+}
+```
+
+<Blockquote mod="warning">
+It's essential to stress that the current environment variables implementation doesn't prevent printing values to the log (or restrict exposing a variable any other way). You should avoid printing sensitive data to the log intentionally.
+</Blockquote>
+
+## Running tests with k6 cloud command
+
+Environment variables set up on the cloud can be referenced on test runs run with `k6 cloud` command. 
+
+### Limitations
+
+Referencing environment variables set in the cloud app in local scripts is possible, but it doesn't work in all cases. If the code in the [init context](https://k6.io/docs/getting-started/running-k6#the-init-context-and-the-default-function) depends on the value of the environment variable, the test run may fail.
```

Comment:
Is this a limitation for all Environment variables or only for Env. variables set on the cloud?

If the former, this limitation should be described at https://k6.io/docs/using-k6/environment-variables.

If the later, we should point out that is a limitation between local and cloud test.

Why? Because one of k6 design goals is not to have differences between the "Execution modes" -> https://k6.io/docs/getting-started/running-k6/#execution-modes

See here one example -> https://k6.io/docs/cloud/creating-and-running-a-test/cloud-tests-from-the-cli/#differences-between-local-and-cloud-execution

Reply:
This limitation could not exist before implementing Env. variables in the cloud, because previously there is no reason to reference an Env. variable that does not exist or not set through CLI flag. Hence I would say this is specific to Env. variables set on the cloud. Yet I'm not sure if this is considered as difference in excution mode as in local script you can't as well reference Env. variables that does not exist. The difference here that if you want to benifit from Env. variables on the cloud when running from CLI (`k6 cloud`), you need to make sure it will be first time evaluated in the cloud as the the value is not available locally, otherwise it [may] fail.

Reply:
> This limitation could not exist before implementing Env. variables in the cloud, because previously there is no reason to reference an Env. variable that does not exist or not set through CLI flag.

You can forget to set the variable through the `-e` flag.

> you need to make sure it will be first time evaluated in the cloud as the the value is not available locally, otherwise it [may] fail.

How does it fail?


I have not been able to test this. Please, can we sync for this one in Slack? 


I think the problem is a known issue of cloud tests that happens independently of using environment variables.

I might be wrong, but I think it was something like:

When running a cloud test, the "init code" is executed before deploying the test on the load generator instances, so a test whose init code is not "error-safe" will fail when deploying the test.

I might be totally wrong but want to double check with the team.

Reply:
Yes you are totally right, this not cloud env specific, rather what you exactly explained. I think I just wanted to stress this in the context of the cloud environemnt variable, but yes it can be consfusing.

Reply:
Then, I suggest removing the limitations section on this page.

This "gotcha" should have a proper error message and document somewhere else on the cloud docs.